pentasharp/Controllers/BusCompanyController.cs

```diff
@@ -2,147 +2,101 @@
-            _context.Buses.UpdateRange(company.Buses);
-
-            _context.SaveChanges();
-
-            return Ok(new { success = true, message = "Company and its buses soft deleted successfully." });
-        }
-
-
-        [HttpPost("AddBus")]
-        public IActionResult AddBus([FromBody] AddBusViewModel model)
+        public async Task<IActionResult> DeleteCompany(int id)
         {
-            if (ModelState.IsValid)
-            {
-                var bus = _mapper.Map<Buses>(model);
-                _context.Buses.Add(bus);
-                _context.SaveChanges();
-                return Ok(new { success = true, message = "Bus added successfully." });
-            }
-            return BadRequest(new { success = false, message = "Invalid data provided." });
-        }
+            var result = await _companyService.DeleteCompanyAsync(id);
+            if (result)
+                return Ok(ResponseFactory.SuccessResponse("Company and its buses deleted successfully.",result));
 
-        [HttpGet("GetBuses")]
-        public IActionResult GetBuses()
-        {
-            var buses = _context.Buses.Include(b => b.BusCompany).ToList();
-            var viewModel = buses.Select(b => new BusViewModel
-            {
-                BusId = b.BusId,
-                BusNumber = b.BusNumber,
-                Capacity = b.Capacity,
-                BusCompanyId = b.BusCompanyId,
-                CompanyName = b.BusCompany.CompanyName
-            }).ToList();
-            return Ok(viewModel);
+            return NotFound(ResponseFactory.ErrorResponse(ResponseCodes.NotFound, ResponseMessages.NotFound));
         }
 
-        [HttpPut("EditBus/{id}")]
-        public IActionResult EditBus(int id, [FromBody] EditBusViewModel model)
+        [HttpGet("GetCompanies")]
+        public async Task<IActionResult> GetCompaniesAsync()
         {
-            var bus = _context.Buses.Find(id);
-            if (bus == null) return NotFound(new { success = false, message = "Bus not found." });
-
-            _mapper.Map(model, bus);
-            _context.SaveChanges();
-            return Ok(new { success = true, message = "Bus updated successfully." });
+            var companies = await _companyService.GetCompaniesAsync();
+            return Ok(companies);
         }
 
-        [HttpDelete("DeleteBus/{id}")]
-        public IActionResult DeleteBus(int id)
+        [HttpGet("GetBusCompanyUser/{companyId}")]
+        public async Task<IActionResult> GetBusCompanyUserAsync(int companyId)
         {
-            var bus = _context.Buses.Find(id);
-            if (bus == null) return NotFound(new { success = false, message = "Bus not found." });
+            var response = await _companyService.GetBusCompanyUserAsync(companyId);
 
-            _context.Buses.Remove(bus);
-            _context.SaveChanges();
-            return Ok(new { success = true, message = "Bus deleted successfully." });
+            if (response.Success)
```

Comment:
try to avoid nested ifs when possible. A case like this can be handled similarly 
if (response.Success)
{
    return Ok(ResponseFactory.SuccessResponse(ResponseMessages.Success, response.Data));
}

var errorMessage = response.Message.ToLower();
if (errorMessage.Contains("not found"))
{
    return NotFound(ResponseFactory.ErrorResponse(ResponseCodes.NotFound, response.Message));
}

return BadRequest(ResponseFactory.ErrorResponse(ResponseCodes.InvalidOperation, response.Message));